src/plugins/condition/Condition.js

```diff
@@ -57,29 +56,41 @@ export default class ConditionClass extends EventEmitter {
         this.conditionManager = conditionManager;
         this.id = conditionConfiguration.id;
         this.criteria = [];
-        this.criteriaResults = {};
         this.result = undefined;
-        this.latestTimestamp = {};
         this.timeSystems = this.openmct.time.getAllTimeSystems();
         if (conditionConfiguration.configuration.criteria) {
             this.createCriteria(conditionConfiguration.configuration.criteria);
         }
         this.trigger = conditionConfiguration.configuration.trigger;
-        this.conditionManager.on('broadcastTelemetry', this.handleBroadcastTelemetry, this);
     }
 
-    handleBroadcastTelemetry(datum) {
+    getResult(datum) {
         if (!datum || !datum.id) {
             console.log('no data received');
             return;
         }
+        if (!this.isTelemetryUsed(datum.id)) {
+            return;
+        }
         this.criteria.forEach(criterion => {
-            if (criterion.telemetry && (criterion.telemetry === 'all' || criterion.telemetry === 'any')) {
-                criterion.handleSubscription(datum, this.conditionManager.telemetryObjects);
+            if (this.isAnyOrAllTelemetry(criterion)) {
+                criterion.getResult(datum, this.conditionManager.telemetryObjects);
             } else {
-                criterion.emit(`subscription:${datum.id}`, datum);
+                criterion.getResult(datum);
             }
         });
+
+        this.result = evaluateResults(this.criteria.map(criterion => criterion.result), this.trigger);
+    }
+
+    isAnyOrAllTelemetry(criterion) {
+        return (criterion.telemetry && (criterion.telemetry === 'all' || criterion.telemetry === 'any'));
+    }
+
+    isTelemetryUsed(id) {
+        return this.criteria.some(criterion => {
+            return this.isAnyOrAllTelemetry(criterion) || criterion.telemetryObjectIdAsString === id;
```

Comment:
`criterion.telemetryObjectIdAsString` was removed. So this will not match

Reply:
wow good catch. i'll make sure to add that back